system-parachains/collectives/collectives-polkadot/src/secretary/mod.rs

```diff
@@ -0,0 +1,294 @@
+
+/// The Secretary members' ranks.
+pub mod ranks {
+	use pallet_ranked_collective::Rank;
+
+	pub const SECRETARY_CANDIDATE: Rank = 0;
+	pub const SECRETARY: Rank = 1;
+}
+
+/// Origins of:
+/// - Root;
+/// - FellowshipAdmin (i.e. token holder referendum);
+/// - Plurality vote from Fellows can promote, demote, remove and approve rank retention
+/// of members of the Secretary Collective (rank `2`).
+type ApproveOrigin = EitherOf<
+	EnsureRootWithSuccess<AccountId, ConstU16<65535>>,
+	EitherOf<
+		MapSuccess<
+			EnsureXcm<IsVoiceOfBody<GovernanceLocation, FellowshipAdminBodyId>>,
+			Replace<ConstU16<2>>,
+		>,
+		MapSuccess<Fellows, Replace<ConstU16<2>>>,
+	>,
+>;
+
+/// Origins of:
+/// - Root;
+/// - FellowshipAdmin (i.e. token holder referendum);
+/// - Secretary;
+/// - Plurality vote from Fellows can kill and cancel proposals.
+/// of members of the Secretary Collective.
+type OpenGovOrSecretaryOrFellow = EitherOfDiverse<
+	EitherOfDiverse<EnsureRoot<AccountId>, Fellows>,
+	EitherOfDiverse<Secretary, EnsureXcm<IsVoiceOfBody<GovernanceLocation, FellowshipAdminBodyId>>>,
+>;
+
+/// Origins of:
+/// - Root;
+/// - FellowshipAdmin (i.e. token holder referendum);
+/// - Plurality vote from Fellows can exchange origins and
+/// - configure the parameters that govern the Collective.
+type OpenGovOrFellow = EitherOfDiverse<
+	EnsureRoot<AccountId>,
+	EitherOfDiverse<Fellows, EnsureXcm<IsVoiceOfBody<GovernanceLocation, FellowshipAdminBodyId>>>,
+>;
+
+impl pallet_secretary_origins::Config for Runtime {}
+
+pub type SecretaryReferendaInstance = pallet_referenda::Instance3;
+
+impl pallet_referenda::Config<SecretaryReferendaInstance> for Runtime {
+	type WeightInfo = (); // TODO weights::pallet_referenda_secretary_referenda::WeightInfo<Runtime>;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeEvent = RuntimeEvent;
+	type Scheduler = Scheduler;
+	type Currency = Balances;
+	// Secretary collective can submit proposals.
+	type SubmitOrigin = pallet_ranked_collective::EnsureMember<
+		Runtime,
+		SecretaryCollectiveInstance,
+		{ ranks::SECRETARY },
+	>;
+	type CancelOrigin = OpenGovOrSecretaryOrFellow;
+	type KillOrigin = OpenGovOrSecretaryOrFellow;
+	type Slash = ToParentTreasury<PolkadotTreasuryAccount, LocationToAccountId, Runtime>;
+	type Votes = pallet_ranked_collective::Votes;
+	type Tally = pallet_ranked_collective::TallyOf<Runtime, SecretaryCollectiveInstance>;
+	type SubmissionDeposit = ConstU128<0>;
+	type MaxQueued = ConstU32<100>;
+	type UndecidingTimeout = ConstU32<{ 7 * DAYS }>;
+	type AlarmInterval = ConstU32<1>;
+	type Tracks = tracks::TracksInfo;
+	type Preimages = Preimage;
+}
+
+pub type SecretaryCollectiveInstance = pallet_ranked_collective::Instance3;
+
+impl pallet_ranked_collective::Config<SecretaryCollectiveInstance> for Runtime {
+	type WeightInfo = (); // TODO weights::pallet_ranked_collective_secretary_collective::WeightInfo<Runtime>;
+	type RuntimeEvent = RuntimeEvent;
+	type AddOrigin = OpenGovOrFellow;
```

Comment:
IMO we should use everywhere here `ApproveOrigin`.

Reply:
And maybe the fellows should have a special track that requires more approval etc.